org.ekstep.reviewercomments-1.0/editor/plugin.js

```diff
@@ -0,0 +1,130 @@
+                        });
+                        ecEditor.getService('telemetry').error({
+                            "env": 'content',
+                            "stage": ecEditor.getCurrentStage().id,
+                            "action": 'review comments',
+                            "objectid": "",
+                            "objecttype": "",
+                            "err": error.status,
+                            "type": "API",
+                            "data": error,
+                            "severity": "fatal"
+                        });
+                        console.warn('error in fetching review comments: ', error);
+                    }
+
+                });
+            }
+        } else {
+            instance.displayStageComments(instance.comments)
+        }
+    },
+    //Show comments after getting the apiResponse
+    showComments: function (comments) {
+        var sortedComments = {}
+        if (comments) {
+            sortedComments = _.sortBy(comments, function (dateObj) {
+                return new Date(dateObj.createdOn);
+            });
+            this.mapComment(sortedComments);
+        }
+    },
+    //To show the 'No review message to the user
+    displayNoComments: function () {
+        jQuery('a[data-content ="comments"]').removeClass('highlight');
+        ecEditor.jQuery('#reviewerComments').html('<div>' + this.NO_COMMENTS + '</div>');
+    },
+    //Function to format the date 
+    getMonthName: function (date) {
+        var date = new Date(date);
+        return date.toLocaleString('en-us', {
+            month: "long"
+        }).toString().substr(0, 3) + " " + date.getDate();
+    },
+    //To map the comments from the apiResponse
+    mapComment: function (sortedComments) {
+        var instance = this;
+        var displayComments = [];
+        var result = _.map(sortedComments, function (item) {
+            var commentThread = {};
+            commentThread.username = item.userInfo.name
+            commentThread.logo = item.userInfo.logo ? item.userInfo.logo : ecEditor.resolvePluginResource(instance.manifest.id, instance.manifest.ver, "assets/reviewer_icon.png");
+            commentThread.username = item.userInfo.name;
+            commentThread.date = instance.getMonthName(item.createdOn);
+            commentThread.body = item.body
+            commentThread.stageId = item.stageId
+            displayComments.push(commentThread);
+        });
+        instance.comments = displayComments;
+        //update the mapped comments to the template      
+        instance.displayStageComments(displayComments);
+    },
+    checkFilter: function (comments) {
+        var filterComments = _.filter(comments, ['stageId', ecEditor.getCurrentStage().id]);
+        return filterComments;
+    },
+    displayStageComments: function (displayComments) {
+        filteredComments = [];
+        filteredComments = this.checkFilter(displayComments)
```

Comment:
checkFilter() function name is not appropriate. It is filterComments()

Reply:
Done